### ISSUE TITLE
Template string fails to populate subfield in certain constellations

### ISSUE THREAD

**Issue by JanRomero** [processwire/processwire-issues#1153]
### Short description of the issue
Forgive me, I’m posting an issue again…

I’m trying to populate a “template string” that includes "{title|parent.title}". For pages with no title, this tag is not populated even if the page’s parent has a title.

### Expected behavior
The parent’s title should be filled in.

### Actual behavior
Once we leave `WireTextTools->populatePlaceholders`, it’s a wild ride through the Page class starting in `getMarkup()`, where the “|” is detected and we’re sent to `getFieldFirstValue()`.

Here, everything is still going according to plan: first it checks if `getUnformatted()` succeeds for any of the keys, and indeed it finds the parent’s title. However, because we want output formatting, it then calls `get()`, which detects the “.” and calls `getFieldSubfieldValue()`.

Here again, things would go smoothly without output formatting. In that case it would call `getDot()` (ends up in `WireData::_getDot()`) which works as expected. With output formatting on, it calls `getField()`, which only works for normal fields:

	// allow limited access to field.subfield properties when output formatting is on
	// we only allow known custom fields, and only 1 level of subfield
	$keys = explode('.', $key);
	$key1 = $keys[0];
	$key2 = $keys[1];
	$field = $this->getField($key1);
	
	if(!$field || ($field->flags & Field::flagSystem)) return null;

If it didn’t return null at that point, it would continue to call `get('parent')` and then `get('title')` on that, and it would work fine.

Apparently this behavior is intended for security reasons. Unfortunately I don’t really grasp the implications of making changes here, or the reason why “we only allow known custom fields”, so I don’t dare make any suggestions, but the behavior is not what you would expect, and it doesn’t seem to be documented. In any case, template strings work fine with parent.title, just not when an OR is involved.

### Steps to reproduce the issue

1. Find a page that has no title but a parent with a title, or delete the title, or use a different empty field in step 2
2. echo `each('{title|parent.title}')` on a PageArray containing that page
3. The page’s parent’s title is not echoed.

### Setup/Environment
* ProcessWire version: 3.0.152

**Comment by ryancramerdesign** [comment]
@JanRomero I can't seem to duplicate it here. I'm using this as a test:
~~~~~
$items = $pages->find("template=role, include=all");
echo "<pre>" . $items->each("{title|parent.title}\n");
~~~~~
Since page template "role" has no "title" field, I get a list of the parent.title for each role:
~~~~~
Roles
Roles
Roles
Roles
Roles
Roles
Roles
Roles
Roles
Roles
~~~~~
Let me know if I'm missing something in terms of how to test it. I'm also wondering if your title field is multi-language or not?

**Comment by JanRomero** [comment]
Hi, thanks, it must have been fixed somewhere between 3.0.172 and the current dev branch. I just tried all possible combinations on 3.0.172 and could reproduce it every time, but updating PW fixed it!
